bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java

```diff
@@ -249,16 +238,104 @@ public void initialize() {
     @Override
     public void dispose() {
         try {
-            BlueGigaSerialHandler bgh = getBgHandler();
-            bgh.removeEventListener(this);
-            bgh.removeHandlerListener(this);
-            bgh.close();
+            stopScheduledTasks();
+            transactionManager.removeEventListener(this);
+            serialHandler.removeHandlerListener(this);
+            transactionManager.close();
+            serialHandler.close();
+            initComplete = false;
+            devices.clear();
+            connections.clear();
         } catch (IllegalStateException e) {
             // ignore if handler wasn't set at all
         }
         closeSerialPort();
     }
 
+    private void schedulePassiveScan() {
+        cancelScheduledPassiveScan();
+        passiveScanIdleTimer = executor.schedule(() -> {
+            if (!activeScanEnabled) {
+                logger.debug("Activate passive scan");
+                bgEndProcedure();
+                bgStartScanning(false, configuration.passiveScanInterval, configuration.passiveScanWindow);
+            } else {
+                logger.debug("Ignore passive scan activation as active scan is active");
+            }
+        }, configuration.passiveScanIdleTime, TimeUnit.MILLISECONDS);
```

Comment:
Rather than constantly cancelling and rescheduling this, why not have it run as a fixed schedule task? Then use locks to make sure that the scheduled task doesn't run when other commands are running. Maybe just synchronize all of the `bg*` methods?

Reply:
Expanding on the locks idea, you could use a ReadWriteLock and have all the `bg*` methods do reads while the passiveScanIdleTimer uses the write lock.

Reply:
Bridge doesn't know is passive scan is running, so basically bridge should always send command to bluegiga controller when scheduled task is triggered. So it will cause more load than just cancel and rescheduled the timer when commands are send.

Reply:
But there has to be a way to refactor it better than wrapping everything in try/finally. Could you do it as part of a `sendTransaction` call?